File: mycollab-web/src/main/java/com/esofthead/mycollab/module/project/events/ProjectMemberEvent.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.esofthead.mycollab.module.project.events;

import java.util.List;

import com.esofthead.mycollab.eventmanager.ApplicationEvent;

/**
 * 
 * @author haiphucnguyen
 */
public class ProjectMemberEvent {

	public static class InviteProjectMembers extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		private List<String> inviteEmails;
		private Integer roleId;

		public InviteProjectMembers(Object source, List<String> emails,
				Integer roleId) {
			super(source, null);
			this.inviteEmails = emails;
			this.roleId = roleId;
		}

		public List<String> getInviteEmails() {
			return inviteEmails;
		}

		public Integer getRoleId() {
			return roleId;
		}
	}

	public static class Search extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public Search(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoList extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoList(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoInviteMembers extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoInviteMembers(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoRead extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoRead(Object source, Object data) {
			super(source, data);
		}
	}

	public static class GotoEdit extends ApplicationEvent {
		private static final long serialVersionUID = 1L;

		public GotoEdit(Object source, Object data) {
			super(source, data);
		}
	}
}

File: mycollab-web/src/main/scala/com/esofthead/mycollab/module/project/events/ProjectMemberEvent.scala
package com.esofthead.mycollab.module.project.events

import com.esofthead.mycollab.eventmanager.ApplicationEvent

import scala.beans.BeanProperty

/**
 * @author MyCollab Ltd.
 * @since 5.0.3
 */
object ProjectMemberEvent {

  class InviteProjectMembers(@BeanProperty var emails: java.util.List[String], @BeanProperty var roleId: Integer,
                             @BeanProperty var inviteMessage: String) extends Serializable {}

  class Search(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoInviteMembers(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}
}
